lib/puppet/pops/loader/module_loaders.rb

```diff
@@ -214,8 +214,12 @@ def initialize(parent_loader, loaders, module_name, path, loader_name, loadables
     end
 
     def existing_path(effective_path)
+      expanded_path = Puppet::FileSystem.expanded_path_string(effective_path)
```

Comment:
The only caller of `expanded_path` now in the code is at https://github.com/Iristyle/puppet/blob/206dfb86b5179f83489c79c1014be2859e24d9e0/lib/puppet/pops/loader/module_loaders.rb#L133

I took a look at the code paths, and saw that I could influence the cached values much earlier in the lifecycle by modifying the `AbstractPathBasedModuleLoader` so that the initializer calls `expand` - means that the various SmartPath instances would have the right style path fed in in the first place, making later checks unnecessary.

Then these modifications can be removed, which I've done in my latest PR update (in addition to fixing broken tests)